src/main/java/com/powsybl/openloadflow/adm/AdmittanceMatrix.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.adm;
+
+import com.powsybl.math.matrix.DenseMatrix;
+import com.powsybl.math.matrix.LUDecomposition;
+import com.powsybl.math.matrix.Matrix;
+import com.powsybl.math.matrix.MatrixFactory;
+import com.powsybl.openloadflow.equations.EquationSystem;
+import com.powsybl.openloadflow.equations.JacobianMatrix;
+import com.powsybl.openloadflow.network.LfBus;
+import org.apache.commons.math3.complex.Complex;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class AdmittanceMatrix implements AutoCloseable {
+
+    private final EquationSystem<AdmittanceVariableType, AdmittanceEquationType> equationSystem;
+
+    private final Matrix matrix;
+
+    private LUDecomposition lu;
+
+    private DenseMatrix e; // impedance extractor
+
+    public AdmittanceMatrix(EquationSystem<AdmittanceVariableType, AdmittanceEquationType> equationSystem, Matrix matrix) {
+        this.equationSystem = Objects.requireNonNull(equationSystem);
+        this.matrix = Objects.requireNonNull(matrix);
+    }
+
+    public static AdmittanceMatrix create(AdmittanceEquationSystem ySystem, MatrixFactory matrixFactory) {
+        Objects.requireNonNull(matrixFactory);
+        try (var j = new JacobianMatrix<>(ySystem.getEquationSystem(), matrixFactory)) {
+            return new AdmittanceMatrix(ySystem.getEquationSystem(), j.getMatrix());
+        }
+    }
+
+    public Matrix getMatrix() {
+        return matrix;
+    }
+
+    /**
+     * Get equivalent impedance between 2 buses.
+     */
+    public Complex getZ(LfBus bus1, LfBus bus2) {
+        Objects.requireNonNull(bus1);
+        Objects.requireNonNull(bus2);
+        var i1xEq = equationSystem.getEquation(bus1.getNum(), AdmittanceEquationType.BUS_ADM_IX).orElseThrow();
+        var i1yEq = equationSystem.getEquation(bus1.getNum(), AdmittanceEquationType.BUS_ADM_IY).orElseThrow();
+        var i2xEq = equationSystem.getEquation(bus2.getNum(), AdmittanceEquationType.BUS_ADM_IX).orElseThrow();
+        var i2yEq = equationSystem.getEquation(bus2.getNum(), AdmittanceEquationType.BUS_ADM_IY).orElseThrow();
+        if (e == null) {
```

Comment:
This is definitely not MT safe.
Can you document the limitation in the class ?

Reply:
Indeed but I don't think it is expected to be thread safe as all of other similar OLF linear algebra utilities.

Reply:
If I didn-t know the underlyings I would expect it to be threadsafe. Maybe a general problem to OLF. 
But running it in MT mode requires a lot of expertis today (see previous PR an MT security analysis).
Anyway, out of scope of ths feature.